src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,160 @@
+import useTheme from "../../hooks/useTheme";
+import useWindow from "../../hooks/useWindow";
+import useResponsivePropsCSS from "../../hooks/useResponsivePropsCSS";
+import { ResponsiveProp } from "../../types";
+import { responsiveSize } from "../../utils/css";
+
+type FooterProps = {
+  children: ReactNode;
+};
+
+function Footer({ children }: FooterProps) {
+  const theme = useTheme();
+
+  return <footer css={{ marginTop: theme.space[8] }}>{children}</footer>;
+}
+
+type Props = {
+  title: string;
+  open: boolean;
+  onClose?: () => void;
+  children: ReactNode;
+  testId?: string;
+} & ResponsiveProp<"width">;
+
+const defaultProps = {
+  open: false,
+  width: "600",
+} as const;
+
+function Modal(props: Props) {
+  const { title, open = false, onClose, children, testId } = props;
+  const windowFromContext = useWindow();
+  const theme = useTheme();
+  const [headerId] = useState(() => `modal-header-${nanoid()}`);
+  const [appElement, setAppElement] = useState<
+    typeof windowFromContext.HTMLElement
+  >();
+  const closeModal = () => {
+    onClose && onClose();
+  };
+  const responsiveCSS = useResponsivePropsCSS(props, defaultProps, {
+    width: responsiveSize("width"),
+  });
+  const responsiveContentCSS = {
+    ...responsiveCSS,
+    position: "absolute",
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    height: "fit-content",
+    border: 0,
+    boxSizing: "border-box",
+    margin: "auto",
+    padding: `${theme.space[4]} ${theme.space[7]} ${theme.space[6]} ${theme.space[7]}`,
+    maxWidth: `calc(100% - ${theme.space[8]})`,
+    backgroundColor: theme.colors.white,
+    borderRadius: theme.radii[1],
+    boxShadow: `
+      rgba(0, 0, 0, 0.1) 0px 10px 15px -3px,
+      rgba(0, 0, 0, 0.05) 0px 4px 6px -2px
+    `,
+    ":focus": {
+      outline: 0,
+    },
+  } as const;
+  const modalStyles: Styles = {
+    overlay: {
+      position: "fixed",
+      top: 0,
+      left: 0,
+      right: 0,
+      bottom: 0,
+      backgroundColor: "rgba(0, 0, 0, 0.5)",
+    },
+  };
+
+  useEffect(() => {
+    const element = windowFromContext.document.querySelector(
+      "[data-basis-modal-app]"
+    );
+
+    if (element) {
+      setAppElement(element);
+    } else {
+      throw new Error(
+        "Modal: could not find an element with the `data-basis-modal-app` attribute."
+      );
+    }
+  }, [windowFromContext]);
+
+  /*
+    Without this, if open={true} by default, react-modal doesn't set 
+    aria-hidden="true" on the appElement.
+  */
+  if (!appElement) {
+    return null;
+  }
+
+  return (
+    <ReactModal
+      css={responsiveContentCSS}
+      style={modalStyles}
+      isOpen={open}
+      onRequestClose={closeModal}
+      appElement={appElement}
+      parentSelector={() => windowFromContext.document.body}
+      contentElement={(
+        props: ComponentPropsWithRef<"div">,
+        children: ReactNode
+      ) => <section {...props}>{children}</section>}
```

Comment:
Maybe have this as an external component, will make it easier to read.

Reply:
ok, will fix